src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs

```diff
@@ -0,0 +1,203 @@
+                }
+            }
+        }
+
+        private async Task RunLogMessagesPump(IWebDriver driver, CancellationToken token)
+        {
+            try
+            {
+                while (!token.IsCancellationRequested)
+                {
+                    PumpLogMessages(driver);
+                    if (driver.CurrentWindowHandle == null) {
+                        // Accessing this property will throw, if the browser has crashed
+                    }
+
+                    await Task.Delay(100, token).ConfigureAwait(false);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // Ignore
+            }
+        }
+
+        private void PumpLogMessages(IWebDriver driver)
+        {
+            foreach (var logEntry in driver.Manage().Logs.GetLog(LogType.Browser))
+            {
+                if (logEntry.Level == SeleniumLogLevel.Severe)
+                {
+                    // throw on driver errors, or other errors that show up
+                    // in the console log
+                    throw new ArgumentException(logEntry.Message);
+                }
+
+                var match = s_consoleLogRegex.Match(Regex.Unescape(logEntry.Message));
+                string msg = match.Success ? match.Groups[1].Value : logEntry.Message;
+                msg += Environment.NewLine;
+
+                _processLogMessage(msg);
+            }
+        }
+
+        private string BuildUrl(string webServerAddr)
+        {
+            var uriBuilder = new UriBuilder($"{webServerAddr}/{_arguments.HTMLFile}");
+            var sb = new StringBuilder();
+            foreach (var arg in _passThroughArguments)
+            {
+                if (sb.Length > 0)
+                    sb.Append("&");
+
+                sb.Append($"arg={HttpUtility.UrlEncode(arg)}");
+            }
+
+            uriBuilder.Query = sb.ToString();
+            return uriBuilder.ToString();
+        }
+
+        private static async Task<string> StartWebServer(string contentRoot, CancellationToken token)
+        {
+            var host = new WebHostBuilder()
+                .UseKestrel()
+                .UseContentRoot(contentRoot)
+                .UseStartup<WasmTestWebServerStartup>()
+                .ConfigureLogging(logging => {
+                    logging.AddConsole().AddFilter(null, LogLevel.Warning);
+                })
+                .UseUrls("http://127.0.0.1:0")
+                .Build();
+
+            await host.StartAsync(token);
+            return host.ServerFeatures
+                    .Get<IServerAddressesFeature>()
+                    .Addresses
+                    .First();
+        }
+
+        private string GetPackagePath ()
+        {
+            try
+            {
+                return _arguments.AppPackagePath;
+            }
+            catch (ArgumentException)
+            {
+                return Directory.GetCurrentDirectory();
```

Comment:
I'm not sure we really need this fallback, requiring passing `--app` is fine imho. It will be passed by the RunTests.sh script in dotnet/runtime anyway.

As a compromise I could live with supporting `--app .` and translating that into `Directory.GetCurrentDirectory()`

Reply:
I agree, leaving it as required parameter will not hurt anything